src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutorExtensions.cs

```diff
@@ -31,7 +32,25 @@ public static async Task<IDelayedException> TryExecuteAsync(this IFunctionExecut
                     logger = loggerFactory.CreateLogger(LogCategories.CreateFunctionCategory(functionInstance.FunctionDescriptor.LogName));
                 }
 
-                functionResult = await executor.TryExecuteAsync(functionInstance, cancellationToken);
+                if (functionInstance is FunctionInstance instance)
+                {
+                    // Build retry context
+                    retryContext = new RetryContext
```

Comment:
RetryContext now contains invocation state that needs to be maintained across retries, so this has to be created once outside the loop, right?